www/source/partials/docs/_using-builder-origin-keys.html.md.erb

```diff
@@ -0,0 +1,129 @@
+
+Creating an origin automatically generates an origin key pair. 
+
+Origin key cryptography is asymmetric: it has a public origin key that you can distribute freely, and a private origin key that you should distribute only to members of the origin.
+
+Habitat Builder and Habitat Depot require access to at least the public origin key to validate packages in the origin. 
+
+Habitat uses the public origin key to verify the integrity of a downloaded Habitat artifact before running it.
+
+By default, Supervisors only run packages for which they have the public origin key, and cannot run packages without the correct origin public origin key.
+You can provide the public origin key to the Supervisors by pointing them to a depot that containing it, or by putting the key on disk outside of Habitat in a location that Habitat knows to look for it, such as `~/.hab/cache/keys` on Linux systems.
+
+## Finding origin keys on Builder
+
+You can find your origin keys on the Builder site by navigating to the relevant origin and selecting the **Keys** tab.
+
+<img src="/images/screenshots/origin-keys.png">
+
+## Finding origin keys locally
+
+You'll find your saved private and public origin keys at `~/.hab/cache/keys` on your host(local) machine and  at `/hab/cache/keys` inside the studio environment. To see your existing origin keys, from your host command line, run:
+
+```shell
+ls -la ~/.hab/cache/keys
+```
+
+```powershell
+Get-ChildItem C:\hab\cache\keys
+```
+
+From within Habitat Studio, run:
+
+```shell
+ls -la ~/hab/cache/keys
+```
+
+```powershell
+Get-ChildItem C:\hab\cache\keys
+```
+
+### Creating origin keys
+
+Creating an origin key pair on on the Habitat Builder site automatically generates two keys, a private origin key and a public origin key. You will use the private origin key to sign your Habitat artifacts and Supervisors will use the public origin key to verify the integrity of these `.hart` files.
+
+You can create origin keys from your host machine or from the Habitat Studio. 
+
+You can create an origin key pair during your initial Habitat installation by following the instructions in the `hab cli setup` interaction.
+If you've already installed Habitat, you can create keys for additional origins by running `hab origin key generate <ORIGIN>`, where <ORIGIN> is the name for your new Habitat origin,  from either the host machine or from within the studio.
+
+To create origin keys from the command line, use:
+
+```bash
+hab origin key generate <ORIGIN>
+```
+
+Your keys will be stored in `~/.hab/cache/keys` on Linux sytems. 
+
+```bash
+ls ~/.hab/cache/keys
+
+test-origin-20190416223046.pub
+test-origin-20190416223046.sig.key
+```
+
+In this case, the origin is named "test-origin". The string of numbers, "20190416223046" show the date and time the key was created, in this case, 2019-04-16-22-30-46. The public key has the file extension `.pub` and the private key has `.sig.key`, which stands for "signing key", another name for the private key.
```

Comment:
```suggestion
In this case, the origin is named "test-origin". The string of numbers, "20190416223046" show the date and time the key was created, in this case, 2019-04-16 22:30:46. The public key has the file extension `.pub` and the private key has `.sig.key`, since the private key is used for cryptographically signing Habitat artifacts.
```